src/thor/triplegbuilder.cc

```diff
@@ -1200,6 +1200,62 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
     trip_edge->set_speed(speed);
   }
 
+  if (controller(kEdgeSpeedsFaded) || controller(kEdgeSpeedsNonFaded)) {
+    // helper function to only get the speed from GetSpeed that we are interested in
+    auto get_speed = [&](uint8_t flow_mask, bool faded,
+                         uint64_t second_of_week = kInvalidSecondsOfWeek) -> std::optional<uint32_t> {
+      uint8_t flow_sources = 0;
+      uint64_t seconds_from_now = 0;
+      uint8_t initial_flow_mask = flow_mask;
+      if (faded) {
+        seconds_from_now = time_info.seconds_from_now;
+        flow_mask |= kCurrentFlowMask;
+      }
+      uint32_t speed = graphtile->GetSpeed(directededge, flow_mask, second_of_week, false,
+                                           &flow_sources, seconds_from_now);
+      if (flow_sources & initial_flow_mask || !initial_flow_mask ||
+          (faded && seconds_from_now == 0)) {
+        return speed;
+      }
+      return std::nullopt;
+    };
+    auto set_speeds = [&](valhalla::TripLeg_Speeds* speeds, bool faded) {
+      std::optional<uint32_t> speed;
+
+      speed = get_speed(kCurrentFlowMask, faded);
```

Comment:
so if `faded == true` and the time is within the fading range, it's probably returning a live speed faded with the _base_ speed right?

Reply:
I can see that you add `kCurrentFlowMask` to get any other `faded` speed type, that's good. but this one might seem strange, as that's most likely not what's happening in a routing algorithm where usually `kPredictedFlowMask` is included and the speed is faded with that. I'd let the data decide here as it'd be done in the routing algorithms and just pass `kDefaultFlowMask` (or maybe even better: respect the `speed_types` costing option?)